openssl.jam doesn't seem to work with the vcpkg openssl

I've made the following simple project to isolate this:
`Jamroot`:
```
using openssl : : <include>"C:/vcpkg/installed/x64-windows/include" <search>"C:/vcpkg/installed/x64-windows/lib" ;

exe main : main.cpp /openssl//crypto ;
```
`main.cpp`:
```
#include <openssl/evp.h>

int main()
{
    OpenSSL_add_all_algorithms();
}
```
When I try to build it, `crypto` is not detected:
```
C:\tmp\test-openssl>b2 -sBOOST_ROOT=C:/boost-git/develop --reconfigure
warning: mismatched versions of B2 engine and core
warning: B2 engine (b2) is 4.4.0
warning: B2 core (at C:/boost-git/develop/tools/build/src) is 4.7-git
Performing configuration checks

    - crypto                   : no
```
and `config.log` contains:
```
compile-c-c++ bin\standalone\ac\msvc-14.3\debug\threading-multi\main-crypto.obj
main-crypto.cpp
msvc.link bin\standalone\ac\msvc-14.3\debug\threading-multi\crypto.exe
LINK : fatal error LNK1181: cannot open input file 'crypto.lib'

        call "bin\standalone\msvc\msvc-14.3\msvc-setup.bat"  >nul
 link /NOLOGO /INCREMENTAL:NO "bin\standalone\ac\msvc-14.3\debug\threading-multi\main-crypto.obj"    "crypto.lib"  /DEBUG /MACHINE:X64 /MANIFEST:EMBED /subsystem:console /out:"bin\standalone\ac\msvc-14.3\debug\threading-multi\crypto.exe" /LIBPATH:"C:\vcpkg\installed\x64-windows\lib" 
```
The name of the library in vcpkg is `libcrypto.lib`, not `crypto.lib`:
```
C:\vcpkg>dir installed\x64-windows\lib
 Volume in drive C has no label.
 Volume Serial Number is 8098-70CA

 Directory of C:\vcpkg\installed\x64-windows\lib

14/09/2021  13:12    <DIR>          .
14/09/2021  13:12    <DIR>          ..
12/09/2021  03:47             6.224 bz2.lib
14/09/2021  13:05         1.011.470 libcrypto.lib
14/09/2021  13:03            17.934 libexpat.lib
14/09/2021  13:03             8.186 libffi.lib
14/09/2021  13:05           121.986 libssl.lib
12/09/2021  03:47            23.422 lzma.lib
04/11/2019  11:31           321.236 msmpi.lib
14/09/2021  13:03    <DIR>          pkgconfig
14/09/2021  13:08           361.556 python39.lib
14/09/2021  13:07            64.078 sqlite3.lib
12/09/2021  03:47            16.334 zlib.lib
12/09/2021  03:42            42.848 zstd.lib
              11 File(s)      1.995.274 bytes
```
I've tried to add `libcrypto` to https://github.com/bfgroup/b2/blob/84ed4e1b0a905d434b422d876c56e07beb2b5a40/src/tools/openssl.jam#L29 but there was no effect.

--- a/src/tools/openssl.jam
+++ b/src/tools/openssl.jam
@@ -43,8 +43,10 @@ if --debug-configuration in [ modules.peek : ARGV ]
 #
 #   <search>
 #       The directory containing the openssl binaries.
-#   <name>
-#       Overrides the default library name.
+#   <ssl-name>
+#       Overrides the default name of ssl library.
+#   <crypto-name>
+#       Overrides the default name of crypto library.
 #   <include>
 #       The directory containing the openssl headers.
 #
@@ -86,9 +88,10 @@ rule init (
 
     local library-path = [ feature.get-values <search> : $(options) ] ;
     local include-path = [ feature.get-values <include> : $(options) ] ;
-    local library-name = [ feature.get-values <name> : $(options) ] ;
+    local ssl-name = [ feature.get-values <ssl-name> : $(options) ] ;
+    local crypto-name = [ feature.get-values <crypto-name> : $(options) ] ;
 
-    if ! $(library-path) && ! $(include-path) && ! $(source-path) && ! $(library-name)
+    if ! $(library-path) && ! $(include-path) && ! $(source-path) && ! $(ssl-name) && ! $(crypto-name)
     {
         is-default = true ;
     }
@@ -122,15 +125,18 @@ rule init (
             }
         }
 
+        ssl-name ?= $(ssl_names) ;
+        crypto-name ?= $(crypto_names) ;
+
         local ssl_lib = [ new ac-library ssl : $(.project) : $(condition) :
-            $(include-path) : $(library-path) : ssl ] ;
+            $(include-path) : $(library-path) ] ;
         $(ssl_lib).set-header openssl/ssl.h ;
-        $(ssl_lib).set-default-names $(ssl_names) ;
+        $(ssl_lib).set-default-names $(ssl-name) ;
 
         local crypto_lib = [ new ac-library crypto : $(.project) : $(condition) :
-            $(include-path) : $(library-path) : crypto ] ;
+            $(include-path) : $(library-path) ] ;
         $(crypto_lib).set-header openssl/crypto.h ;
-        $(crypto_lib).set-default-names $(crypto_names) ;
+        $(crypto_lib).set-default-names $(crypto-name) ;
 
         targets.main-target-alternative $(ssl_lib) ;
         targets.main-target-alternative $(crypto_lib) ;

